### PR TITLE
Update package

### PR DESCRIPTION
**Pull request by maniackcrudelis**
## Problem
- *Update to last standart*

## Solution
- *Upgrade to 4.9.1*
- *Add a change_url script*
- *Add fail2ban*
- *Add ynh_abort_if_up_to_date*
- *Add ynh_send_readme_to_admin*

## PR Status
Work finished. Package_check, basic tests and upgrade from last version OK.  
Could be reviewed and tested.

## Validation
---
*Minor decision*
- [x] **Upgrade previous version** : Verified by the CI.
- [x] **Code review** : JimboJoe
- [x] **Approval (LGTM)** : JimboJoe
- [ ] **Approval (LGTM)** : 
- [x] **CI succeeded** : [Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/wordpress_ynh%20update_package%20(Official)/)  
When the PR is mark as ready to merge, you have to wait for 3 days before really merge it.

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -6,9 +6,9 @@
 		"en": "Create a beautiful blog or website easily",
 		"fr": "Logiciel de création de blog ou de site Web"
 	},
-	"version": "4.8",
+	"version": "4.9.1~ynh1",
```
Comment (JimboJoe) [inline manifest.json:9]: Here our new versioning rule gives us "4.9.1-1", doesn't it?

File: scripts/_common.sh [hunk 2]
```diff
@@ -33,3 +33,182 @@ CHECK_SIZE () {	# Vérifie avant chaque backup que l'espace est suffisant
...
+logpath = $logpath
+maxretry = $max_retry" 
+EOF
+
+	sudo tee $finalfail2banfilterconf <<EOF
+[INCLUDES]
+before = common.conf
+[Definition]
+failregex = $failregex
+ignoreregrex ="
```
Comment (JimboJoe) [inline scripts/_common.sh]: Here the helper should be updated (extraneous r)

File: scripts/_common.sh [hunk 3]
```diff
@@ -33,3 +33,182 @@ CHECK_SIZE () {	# Vérifie avant chaque backup que l'espace est suffisant
...
+		done
+		echo "$recipients"
+	}
+	recipients=$(find_mails "$recipients")
+
+	local mail_subject="☁️🆈🅽🅷☁️: \`$app\` was just installed!"
+
+	local mail_message="This is an automated message from your beloved YunoHost server.
+
+Specific informations for the application $app.
```
Comment (JimboJoe) [inline scripts/_common.sh]: Minor typo: information always remains singular 😉

File: scripts/_common.sh [hunk 4]
```diff
@@ -33,3 +33,182 @@ CHECK_SIZE () {	# Vérifie avant chaque backup que l'espace est suffisant
...
+#
+# This helper should be used to avoid an upgrade of a package
+# when it's not needed.
+#
+# To force an upgrade, even if the package is up to date,
+# you have to set the variable YNH_FORCE_UPGRADE before.
+# example: sudo YNH_FORCE_UPGRADE=1 yunohost app upgrade MyApp
+#
+# usage: ynh_abort_if_up_to_date
+ynh_abort_if_up_to_date () {
```
Comment (JimboJoe) [inline scripts/_common.sh:141]: Shouldn't that be implemented in the core...?

**Review by JimboJoe — COMMENTED** [review]
LGTM and review globally OK except small points

File: manifest.json (hunk 1 above)
Comment (maniackcrudelis) [reply manifest.json:9]: That's not what I understood.
Is there anything new about that ?

After what josue (?) said, I understood it was better to use a tilde and "ynh" to not interfere with the version number of the app.

File: scripts/_common.sh (hunk 3 above)
Comment (maniackcrudelis) [reply scripts/_common.sh]: Really !?
So, I trust you ;)

File: scripts/_common.sh (hunk 2 above)
Comment (maniackcrudelis) [reply scripts/_common.sh]: But, where is the updated version ?
There's nothing new here https://github.com/YunoHost-Apps/Experimental_helpers/blob/master/ynh_add_fail2ban_config/ynh_add_fail2ban_config

File: scripts/_common.sh (hunk 4 above)
Comment (maniackcrudelis) [reply scripts/_common.sh:141]: Why not...
But it isn't for now...

File: manifest.json (hunk 1 above)
Comment (JimboJoe) [reply manifest.json:9]: OK, looks like I totally missed [it](https://github.com/YunoHost/doc/commit/25471ccb257bc5ecfb39675a3b5c57821cf57ec5)... so  please forget it 😊

File: scripts/_common.sh (hunk 2 above)
Comment (JimboJoe) [reply scripts/_common.sh]: I proposed changes in [this PR](https://github.com/YunoHost-Apps/Experimental_helpers/pull/2).

File: scripts/_common.sh (hunk 4 above)
Comment (JimboJoe) [reply scripts/_common.sh:141]: OK, just trying not to clutter apps (and complexify them) if we can avoid it... Of course we can merge that, but maybe could we elaborate/discuss further a core enhancement before applying that change to other apps...?

File: scripts/_common.sh (hunk 3 above)
Comment (JimboJoe) [reply scripts/_common.sh]: Such a heavy responsibility on my shoulders... :flushed:

File: scripts/_common.sh (hunk 2 above)
Comment (maniackcrudelis) [reply scripts/_common.sh]: Whoops ! Sorry I missed that !

**Review by JimboJoe — APPROVED** [review]
LGTM and code review OK 👍

**Pull request merged**